linkerd/app/inbound/src/policy/authorize/tcp.rs

```diff
@@ -122,42 +122,39 @@ where
     }
 
     fn call(&mut self, io: I) -> Self::Future {
-        match self {
-            // If the connection is authorized, pass it to the inner service and stop processing the
-            // connection if the authorization's state changes to no longer permit the request.
-            Self::Authorized(Authorized {
-                inner,
-                client,
-                tls,
-                policy,
-                metrics,
-            }) => {
-                let client = *client;
-                let tls = tls.clone();
-                let mut policy = policy.clone();
-                let metrics = metrics.clone();
-
-                // FIXME increment counter.
-
-                let call = inner.call(io);
-                future::Either::Left(Box::pin(async move {
-                    tokio::pin!(call);
-                    loop {
-                        tokio::select! {
-                            res = &mut call => return res.map_err(Into::into),
-                            _ = policy.changed() => {
-                                if let Err(denied) = policy.check_authorized(client, &tls) {
-                                    metrics.terminate(&policy);
-                                    tracing::info!(%denied, "Connection terminated");
-                                    return Err(denied.into());
-                                }
-                            }
-                        };
+        let Authorized {
+            inner,
+            client,
+            tls,
+            policy,
+            metrics,
+        } = match self {
+            Self::Authorized(a) => a,
+            Self::Unauthorized(_deny) => unreachable!("poll_ready must be called"),
+        };
+
+        // If the connection is authorized, pass it to the inner service and stop processing the
+        // connection if the authorization's state changes to no longer permit the request.
+        let client = *client;
+        let tls = tls.clone();
+        let mut policy = policy.clone();
+        let metrics = metrics.clone();
+
+        let call = inner.call(io);
+        future::Either::Left(Box::pin(async move {
+            tokio::pin!(call);
+            loop {
+                tokio::select! {
+                    res = &mut call => return res.map_err(Into::into),
+                    _ = policy.changed() => {
+                        if let Err(denied) = policy.check_authorized(client, &tls) {
+                            tracing::info!(server = %policy.server_label(), ?tls, %client, "Connection terminated");
```

Comment:
IMO "Connection terminated" seems a little vague, I'm not sure if it will be obvious to a reader that this connection was terminated because of policy --- it might just be interpreted as "the connection was closed".

Maybe rephrase it to something like
```suggestion
                            tracing::info!(
                                server = %policy.server_label(),
                                ?tls, 
                                %client,
                                "Connection terminated due to policy change",
                             );
```
or even
IMO "Connection terminated" seems a little vague, I'm not sure if it will be obvious to a reader that this connection was terminated because of policy --- it might just be interpreted as "the connection was closed".

Maybe rephrase it to something like
```suggestion
                            tracing::info!(
                                server = %policy.server_label(),
                                ?tls, 
                                %client,
                                "Connection was no longer authorized after policy change; terminating",
                             );
```